test/Datadog.Trace.TestHelpers/EnvironmentHelper.cs

```diff
@@ -0,0 +1,357 @@
+                "CORECLR_PROFILER",
+                "CORECLR_PROFILER_PATH",
+                "CORECLR_PROFILER_PATH_32",
+                "CORECLR_PROFILER_PATH_64",
+
+                // .NET Framework
+                "COR_ENABLE_PROFILING",
+                "COR_PROFILER",
+                "COR_PROFILER_PATH",
+
+                // Datadog
+                "DD_PROFILER_PROCESSES",
+                "DD_INTEGRATIONS",
+                "DATADOG_PROFILER_PROCESSES",
+                "DATADOG_INTEGRATIONS",
+            };
+
+            foreach (string variable in environmentVariables)
+            {
+                Environment.SetEnvironmentVariable(variable, null);
+            }
+        }
+
+        public void SetEnvironmentVariableDefaults(
+            int agentPort,
+            string processPath,
+            StringDictionary environmentVariables)
+        {
+            if (IsCoreClr())
+            {
+                environmentVariables["CORECLR_ENABLE_PROFILING"] = "1";
+                environmentVariables["CORECLR_PROFILER"] = EnvironmentHelper.ProfilerClsId;
+                environmentVariables["CORECLR_PROFILER_PATH"] = GetProfilerPath();
+            }
+            else
+            {
+                environmentVariables["COR_ENABLE_PROFILING"] = "1";
+                environmentVariables["COR_PROFILER"] = EnvironmentHelper.ProfilerClsId;
+                environmentVariables["COR_PROFILER_PATH"] = GetProfilerPath();
+            }
+
+            environmentVariables["DD_PROFILER_PROCESSES"] = processPath;
+
+            string integrations = string.Join(";", GetIntegrationsFilePaths());
+            environmentVariables["DD_INTEGRATIONS"] = integrations;
+            environmentVariables["DD_TRACE_AGENT_HOSTNAME"] = "localhost";
+            environmentVariables["DD_TRACE_AGENT_PORT"] = agentPort.ToString();
+
+            // for ASP.NET Core sample apps, set the server's port
+            environmentVariables["ASPNETCORE_URLS"] = $"http://localhost:{agentPort}/";
+
+            foreach (var name in new string[] { "REDIS_HOST" })
+            {
+                var value = Environment.GetEnvironmentVariable(name);
+                if (!string.IsNullOrEmpty(value))
+                {
+                    environmentVariables[name] = value;
+                }
+            }
+        }
+
+        public string[] GetIntegrationsFilePaths()
+        {
+            if (_integrationsFileLocation == null)
+            {
+                string fileName = "integrations.json";
+
+                var directory = GetSampleApplicationOutputDirectory();
+
+                var relativePath = Path.Combine(
+                    "profiler-lib",
+                    fileName);
+
+                _integrationsFileLocation = Path.Combine(
+                    directory,
+                    relativePath);
+
+                // TODO: get rid of the fallback options when we have a consistent convention
```

Comment:
If we convert to msbuild throughout this can automatically go away.

Reply:
Not sure if I'd prefer msbuild or cmake for the c++ projects though?

Reply:
Cross-platform would be great, if possible, and VS supports CMake.

Reply:
msbuild is also now cross-plat 😉

Reply:
You're not wrong, but even VS itself uses CMake for cross-plat C++ projects!